state/metadata/state_manager_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package metadata_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/state/metadata"
+)
+
+func TestConflictingPrefixes(t *testing.T) {
+	require := require.New(t)
+
+	metadataManager := metadata.NewDefaultManager()
+
+	// Test no conflicts
+	require.False(
```

Comment:
Can we turn each of these cases into a test struct ie

```golang
type test struct {
	name string
	m MetadataManager
	vmPrefixes [][]byte
	hasConflicts bool
}
```

Reply:
Here's an example of this approach: https://github.com/ava-labs/hypersdk/blob/bca77398df99cce2920fb35feb5abc61f646c6cd/state/keys_test.go#L35

Reply:
Done